lib/cylc/unicode_util.py

```diff
@@ -15,9 +15,20 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-"""Wrap communications daemon for a suite."""
+"""Unicode utility."""
 
-from cylc.network.method import METHOD
 
-if METHOD in ["https", "http"]:
-    from cylc.network.https.daemon import CommsDaemon
+def unicode_encode(data):
```

Comment:
This doesn't encode non-unicode strings, purposeful?

Reply:
This function comes from https://github.com/cylc/cylc/blob/master/lib/cylc/network/https/util.py#L65 and I have not made any logical change to it, so not sure.

Reply:
This line confuses me `{unicode_encode(key): unicode_encode(value)}`.

Why would we try to `unicode_encode` the key if `unicode_encode` doesn't encode strings.

Reply:
Perhaps the purpose of this logic is to ensure any unicode conforms to UTF-8 rather than to encode strings to unicode. Perhaps a name change could alleviate this confusion.

Reply:
Yes - a `unicode` data type is not encoded. `u'my string'.encode()` converts the unicode string into bytes. So if it's already a `str` (which is already bytes) it is skipped, otherwise if it's a `unicode` string it gets encoded.

Reply:
Will rename this `utf8_enforce`.